packages/block-library/src/term-template/edit.js

```diff
@@ -0,0 +1,277 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { memo, useMemo, useState } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import {
+	BlockControls,
+	BlockContextProvider,
+	__experimentalUseBlockPreview as useBlockPreview,
+	useBlockProps,
+	useInnerBlocksProps,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { Spinner, ToolbarGroup } from '@wordpress/components';
+import { useEntityRecords } from '@wordpress/core-data';
+
+const TEMPLATE = [ [ 'core/paragraph' ] ];
+
+function TermTemplateInnerBlocks( { classList, term } ) {
+	const innerBlocksProps = useInnerBlocksProps(
+		{ className: clsx( 'wp-block-term', classList ) },
+		{ template: TEMPLATE, __unstableDisableLayoutClassNames: true }
+	);
+	return <li { ...innerBlocksProps }>{ term?.name }</li>;
+}
+
+function TermTemplateBlockPreview( {
+	blocks,
+	blockContextId,
+	classList,
+	isHidden,
+	setActiveBlockContextId,
+	termName,
+} ) {
+	const blockPreviewProps = useBlockPreview( {
+		blocks,
+		props: {
+			className: clsx( 'wp-block-term', classList ),
+		},
+	} );
+
+	const handleOnClick = () => {
+		setActiveBlockContextId( blockContextId );
+	};
+
+	const style = {
+		display: isHidden ? 'none' : undefined,
+	};
+
+	return (
+		<li
+			{ ...blockPreviewProps }
+			tabIndex={ 0 }
+			// eslint-disable-next-line jsx-a11y/no-noninteractive-element-to-interactive-role
+			role="button"
+			onClick={ handleOnClick }
+			onKeyPress={ handleOnClick }
+			style={ style }
+		>
+			{ termName }
+		</li>
+	);
+}
+
+const MemoizedTermTemplateBlockPreview = memo( TermTemplateBlockPreview );
```

Comment:
It might be good to add a comment explaining why we need the memo.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/70747/commits/ef863fd17f0e3b2dbd441db8679e2646dc7bdca3.